### DISCUSSION TITLE
Manual modified publish of states (mqtt)?

### DISCUSSION THREAD

**Discussion by RaW79** [HASwitchPlate/openHASP#819] [Q&A]
Hi
I know this might not be an openHASP topic but because I´m in need for a solution maybe you can help.

At the moment the mqtt broker or client from the used automation system seems not to be able to handle a payload from (toggle =true) switches or buttons.
So if only the event is published like this:

```{"event" : "up"}```

it works but if in addition an included value like this:

```
{"event" : "up", "val": 1} 
```

it does not work. I checked with tool MQTT Explorer and new seperate MQTT Broker (Mosquitto) which are working fully fine, but my used automation system is the problem.
Because it will need some time if hopefully a developer will fix [this](https://github.com/ioBroker/ioBroker.mqtt/pull/467), I´m looking for a workaround.

Is it possible to publish a modified payload?

e.g. p1b5 is publich automatically by openHASP
```
{"event" : "up", "val": 1} 
```
Can you suggest a code to add and where to publish sth like this (only the value in the payload) in addition?
```
if p1b5 is changed
 publish hasp/plate35/command/Man_p1b5 {%value%}
endif
```
Thank you very much and sorry for this maybe a bit strange request....because I´m in need for a working display solution.

**Comment by hb020** [comment]
You could intercept the outgoing messages in `custom_state_subtopic`, as it gets copies of outgoing messages, and then use `mqtt_send_state` to send out an additional modified version. But frankly, an external stream processor (meaning: an external program to read from a topic and publish a modified version of that message on another topic) would be best, as this problem is not your device's problem, but a problem of your downstream system. Why bother your device with it.

**Reply by RaW79** [reply]
Thanks for your feedback and recommendation. Because I´m not used to build such solutions but based on this hints I can start and give it a try. Thx!
